tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Hierarchical.cpp

```diff
@@ -0,0 +1,2946 @@
+  // Use non-unity (because that's the default) but close alpha values to make
+  // the math easier but still test thoroughly.
+  Limiter hierarchical{
+      HierarchicalType::Krivodonova,
+      std::vector<double>(
+          Spectral::maximum_number_of_points<Spectral::Basis::Legendre>, 0.99)};
+
+  NeighborData<dim, typename Limiter::PackagedData> neighbor_data{};
+
+  const Element<dim> element(ElementId<dim>{0}, {});
+  // We don't care about the ElementId for these tests, just the direction.
+  Limiter::PackagedData& package_data_up_xi = neighbor_data[std::make_pair(
+      Direction<dim>::upper_xi(), ElementId<dim>{0})];
+  Limiter::PackagedData& package_data_lo_xi = neighbor_data[std::make_pair(
+      Direction<dim>::lower_xi(), ElementId<dim>{0})];
+
+  package_data_up_xi.modal_volume_data.initialize(num_pts);
+  package_data_up_xi.mesh = mesh;
+  package_data_lo_xi.modal_volume_data.initialize(num_pts);
+  package_data_lo_xi.mesh = mesh;
+
+  Scalar<DataVector> nodal_scalar_data_to_limit(num_pts, 0.0);
+  tnsr::I<DataVector, dim> nodal_vector_data_to_limit(num_pts, 0.0);
+  Scalar<DataVector> nodal_scalar_expected(num_pts, 0.0);
+  tnsr::I<DataVector, dim> nodal_vector_expected(num_pts, 0.0);
+
+  assemble_from_legendre_modal_coefficients(&get(nodal_scalar_data_to_limit),
+                                            mesh, x, {3.0, -2.0, 2.0, 1.0});
+  get(get<::Tags::Modal<ScalarTag<0>>>(package_data_up_xi.modal_volume_data)) =
+      ModalVector{7.0, 3.0, 2.5, 0.0};
+  get(get<::Tags::Modal<ScalarTag<0>>>(package_data_lo_xi.modal_volume_data)) =
+      ModalVector{1.0, -5.0, 1.0e-18, 0.0};
+  assemble_from_legendre_modal_coefficients(&get(nodal_scalar_expected), mesh,
+                                            x, {3.0, -2.0, 2.0, 0.5 * 0.99});
+
+  assemble_from_legendre_modal_coefficients(&nodal_vector_data_to_limit.get(0),
+                                            mesh, x, {3.0, 2.0, 2.0, 1.0});
+  get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_up_xi.modal_volume_data)
+      .get(0) = ModalVector{7.0, 5.0, 2.5, 0.0};
+  get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_lo_xi.modal_volume_data)
+      .get(0) = ModalVector{-2.1, 1.1, 1.0e-18, 0.0};
+  assemble_from_legendre_modal_coefficients(&nodal_vector_expected.get(0), mesh,
+                                            x,
+                                            {3.0, 2.0, 0.9 * 0.99, 0.5 * 0.99});
+
+  hierarchical(&nodal_scalar_data_to_limit, &nodal_vector_data_to_limit,
+               element, mesh, neighbor_data);
+  CHECK_ITERABLE_APPROX(get(nodal_scalar_data_to_limit),
+                        get(nodal_scalar_expected));
+  for (size_t i = 0; i < dim; ++i) {
+    CAPTURE(i);
+    CHECK_ITERABLE_APPROX(nodal_vector_data_to_limit.get(i),
+                          nodal_vector_expected.get(i));
+  }
+}
+
+void run() noexcept {
+  INFO("Testing 1d limiter");
+  for (size_t order = 2; order < 7; ++order) {
```

Comment:
you can use the `Spectral::minimum_number_of_points` and `Spectral::maximum_number_of_points` here